File: library/src/main/java/me/shkschneider/skeleton/data/MemoryCache.java
package me.shkschneider.skeleton.data;

import android.support.annotation.IntRange;
import android.support.v4.util.LruCache;

public final class MemoryCache<K, V> extends LruCache<K, V> {

    public MemoryCache() {
        this(42);
    }

    public MemoryCache(@IntRange(from=0) final int maxSize) {
        super(maxSize);
    }

    @Override
    protected int sizeOf(final K key, final V value) {
        return 1;
    }

    // put()

    // get()

    synchronized public void clear() {
        evictAll();
    }

}

File: library/src/main/kotlin/me/shkschneider/skeleton/data/MemoryCache.kt
package me.shkschneider.skeleton.data

import android.support.annotation.IntRange
import android.support.v4.util.LruCache

class MemoryCache<K, V> : LruCache<K, V> {

    constructor(@IntRange(from = 0) maxSize: Int = 42) : super(maxSize)

    override fun sizeOf(key: K?, value: V?): Int {
        return 1
    }

    // put()

    // get()

    @Synchronized
    fun clear() {
        evictAll()
    }

}